File: src/rmcs_pubsub/rmcs_pubsub/subscriber.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class Subscriber(Node): # 创建叫subcriber的节点
    def __init__(self):
        super().__init__('subscriber')
        self.create_subscription(String, 'rmcs_status', self.on_message, 10) # 订阅发布者

    def on_message(self, message):
        self.get_logger().info(f'Received: {message.data}') # 接受成功打印信息


def main(args=None):
    rclpy.init(args=args)
    node = Subscriber()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        if rclpy.ok():
            rclpy.shutdown()

File: src/rmcs_pubsub/src/subscriber.cpp
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class Subscriber : public rclcpp::Node
{
public:
  Subscriber()
  : Node("subscriber")
  {
    subscription_ = create_subscription<std_msgs::msg::String>(
      "rmcs_status", 10,
      [this](std_msgs::msg::String::SharedPtr message) {
        RCLCPP_INFO(get_logger(), "Received: %s", message->data.c_str());
      });
  }

private:
  rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<Subscriber>());
  rclcpp::shutdown();
  return 0;
}
